### PR TITLE
Cache api instance

### PR DESCRIPTION
**Pull request by psykzz**

### CODE REVIEW

**Comment by psykzz** [comment]
@DEKHTIARJonathan Added that for you.

**Comment by psykzz** [comment]
Any reason not to merge this?

File: pyup/providers/github.py [hunk 1]
```diff
@@ -12,13 +12,16 @@ class Provider(object):
...
+        self.__api = None
 
     @classmethod
     def is_same_user(cls, this, that):
         return this.login == that.login
 
     def _api(self, token):
-        return Github(token, timeout=50)
+        if not self.__api:
+            self.__api = Github(token, timeout=50)
```
Comment (rafaelpivato) [inline pyup/providers/github.py]: Need to check if **token** changed as well. In that case you would need to get new instance again.

**Comment by psykzz** [comment]
Will pick this up again today and try to get it updated. When im finished i'll squash my commits again before merging.

**Review by rafaelpivato — APPROVED** [review]
This looks awesome! Thanks @psykzz :sunflower:

**Pull request merged**